user_tools/src/spark_rapids_tools/storagelib/csppath.py

```diff
@@ -308,16 +312,52 @@ def create_dirs(self, exist_ok: bool = True):
             # check that the file does not exist
             if self.exists():
                 raise CspFileExistsError(f'Path already Exists: {self}')
-        self.fs_obj.create_dir(self.no_prefix)
+        self.fs_obj.create_dir(self.no_scheme)
         # force the file information object to be retrieved again by invalidating the cached property
         if 'file_info' in self.__dict__:
             del self.__dict__['file_info']
 
     def open_input_stream(self):
-        return self.fs_obj.open_input_stream(self.no_prefix)
+        return self.fs_obj.open_input_stream(self.no_scheme)
 
     def open_output_stream(self):
-        return self.fs_obj.open_output_stream(self.no_prefix)
+        return self.fs_obj.open_output_stream(self.no_scheme)
+
+    def create_sub_path(self, relative: str) -> 'CspPath':
+        """
+        Given a relative path, it will return a new CspPath object with the relative path appended to
+        the current path. This is just for building a path, and it does not call mkdirs.
+        For example,
+        ```py
+        root_folder = CspPath('gs://bucket-name/folder_00/subfolder_01')
+        new_path = root_folder.create_sub_path('subfolder_02')
```

Comment:
just curious does this handle subpath like "../../subfolder_02"?  The reason I ask is that is what I think of when I see relative path, if not maybe we just reword about appending path

Reply:
Good catch..
I will file separate issue to handle `../` in path.